ReactNativeClient/lib/services/KeymapService.ts

```diff
@@ -0,0 +1,202 @@
+const BaseService = require('lib/services/BaseService');
+const { Logger } = require('lib/logger.js');
+const { shim } = require('lib/shim.js');
+
+class KeymapService extends BaseService {
+	private keymap: { [command: string]: KeymapItem };
+
+	static modifiers = /^(CommandOrControl|Command|Control|Ctrl|Cmd|CmdOrCtrl|Alt|Option|AltGr|Shift|Super)$/;
+	static keyCodes = /^([0-9A-Z)!@#$%^&*(:+<_>?~{|}";=,\-./`[\\\]']|F1*[1-9]|F10|F2[0-4]|Plus|Space|Tab|Backspace|Delete|Insert|Return|Enter|Up|Down|Left|Right|Home|End|PageUp|PageDown|Escape|Esc|VolumeUp|VolumeDown|VolumeMute|MediaNextTrack|MediaPreviousTrack|MediaStop|MediaPlayPause|PrintScreen)$/;
+
+	static defaultKeymap: KeymapItem[] = [
+		{ accelerator: 'CommandOrControl+N', command: 'newNoteItem', platforms: ['darwin', 'win32', 'linux', 'freebsd'] },
+		{ accelerator: 'CommandOrControl+T', command: 'newTodoItem', platforms: ['darwin', 'win32', 'linux', 'freebsd'] },
+		{ accelerator: 'CommandOrControl+S', command: 'synchronise', platforms: ['darwin', 'win32', 'linux', 'freebsd'] },
+		{ accelerator: 'CommandOrControl+P', command: 'printItem', platforms: ['darwin', 'win32', 'linux', 'freebsd'] },
+		{ accelerator: 'CommandOrControl+H', command: 'hideJoplin', platforms: ['darwin'] },
+		{ accelerator: 'CommandOrControl+Q', command: 'quit', platforms: ['darwin', 'win32', 'linux', 'freebsd'] },
+		{ accelerator: 'CommandOrControl+,', command: 'preferences', platforms: ['darwin'] },
+		{ accelerator: 'Command+W', command: 'closeWindow', platforms: ['darwin'] },
```

Comment:
What does "closeWindow" do? Exit the app completely or just hide the window?

Reply:
Hide and Close window seems to be the same. But there are two menu items. I really don't know why..